projects/cdk/date-time/time.ts

```diff
@@ -134,6 +135,22 @@ export class TuiTime implements TuiTimeLike {
         );
     }
 
+    private static parseHours(time: string): number {
+        const hours = Number(time.slice(0, 2));
+        const [meridiem = ''] =
+            /[ap]m/.exec(time.toLowerCase().replaceAll(/\W/g, '')) || [];
```

Comment:
Maybe take it out to variable with "talking name" such complicated regexp? What do you think?

Reply:
I've created new static method `parseMeridiemPeriod`

Reply:
✅